drivers/hmis/app/jobs/hmis/activity_log_processor_job.rb

```diff
@@ -0,0 +1,153 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# = Hmis ActivityLogProcessorJob
+#
+# Link the access logs for graphql object/fields to database ids for root entity types (client, enrollment, project).
+# This linking is done async to avoid overhead of doing it inside a request
+#
+module Hmis
+  class ActivityLogProcessorJob < ::BaseJob
+    def perform(force: false)
+      Hmis::ActivityLog.with_advisory_lock(
+        'AccessLogProcessorLock',
+        timeout_seconds: 0, # don't wait
+      ) do
+        scope = Hmis::ActivityLog.all
+        scope = scope.unprocessed unless force
+        process_records(scope)
+      end
+    end
+
+    protected
+
+    RESOLVE_ENROLLMENT_IDS = ->(enrollment_ids) {
+      e_t = Hmis::Hud::Enrollment.arel_table
+      c_t = Hmis::Hud::Client.arel_table
+      client_ids_map = Hmis::Hud::Enrollment.with_deleted.
+        where(id: enrollment_ids).
+        joins(
+          e_t.create_join(
+            c_t,
+            e_t.create_on(
+              # joins to clients; intentionally includes deleted records
+              [e_t[:PersonalID].eq(c_t[:PersonalID]), e_t[:data_source_id].eq(c_t[:data_source_id])].inject(&:and),
+            ),
+          ),
+        ).
+        pluck(e_t[:id], c_t[:id]).to_h
+
+      enrollment_ids.map do |enrollment_id|
+        {
+          enrollment_ids: [enrollment_id],
+          client_ids: [client_ids_map[enrollment_id&.to_i]],
+        }
+      end
+    }
+    # Batch resolve graphql references. Must return 1:1 mapping from ids to entities
+    # Note:
+    #   * There are many more entities we could resolve against but we only resolve these few to reduce maintenance
+    #     burden. It is assumed that to access PII, a query would have to traverse one of the entities below.
+    # graphql_name => ->(unique_id_strings) {
+    #   [
+    #     {client_ids: [], enrollment_ids:[] }
+    #   ]
+    # }
+    RESOLVERS = {
+      'Assessment' => ->(assessment_ids) {
+        e_t = Hmis::Hud::Enrollment.arel_table
+        cas_t = Hmis::Hud::CustomAssessment.arel_table
+        enrollment_id_map = Hmis::Hud::CustomAssessment.with_deleted.
+          where(id: assessment_ids).
+          joins(
+            cas_t.create_join(
+              e_t,
+              cas_t.create_on(
+                # joins to enrollments; intentionally includes deleted records
+                [
+                  e_t[:EnrollmentID].eq(cas_t[:EnrollmentID]),
+                  e_t[:PersonalID].eq(cas_t[:PersonalID]), # this seems redundant
+                  e_t[:data_source_id].eq(cas_t[:data_source_id]),
+                ].inject(&:and),
```

Comment:
same here, I'm pretty sure a merge with the `with_deleted` scopes will give you the same result.